### PR TITLE
Refactor FXIOS-7812 [v121] Fakespot - iPad - sidebar open/close state resets itself when user changes from landscape to portrait mode

### PR DESCRIPTION
**Pull request by thatswinnie**
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-7812), [Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-7813), [Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-7815)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/17419), [Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/17421), [Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/17423)

## :bulb: Description
Refactor to addresse a multiple Fakespot sidebar bugs. We now use Redux to open/close the sidebar and remember if the sidebar should be re-opened.

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

### CODE REVIEW

File: Client/Frontend/Toolbar+URLBar/TabLocationView.swift [hunk 1]
```diff
@@ -290,7 +289,13 @@ class TabLocationView: UIView, FeatureFlaggable {
     @objc
     func didPressShoppingButton(_ button: UIButton) {
         button.isSelected = true
-        delegate?.tabLocationViewDidTapShopping(self, button: button)
+        TelemetryWrapper.recordEvent(category: .action, method: .tap, object: .shoppingButton)
+
+        guard let fakespotState = store.state.screenState(FakespotState.self, for: .fakespot) else {
```
Comment (yoanarios) [inline Client/Frontend/Toolbar+URLBar/TabLocationView.swift:294]: I think the usage of the fakespotState directly from the store is justified here I just wanted to double-check with @OrlaM

File: Client/Frontend/Fakespot/FakespotUtils.swift [hunk 2]
```diff
@@ -88,8 +88,12 @@ public struct FakespotUtils: FeatureFlaggable {
     func shouldDisplayInSidebar(device: UIUserInterfaceIdiom = UIDevice.current.userInterfaceIdiom,
```
Comment (lmarceau) [inline Client/Frontend/Fakespot/FakespotUtils.swift:88]: Can we have unit tests for this?

Comment (thatswinnie) [reply Client/Frontend/Fakespot/FakespotUtils.swift:88]: We already have tests for that.

Comment (lmarceau) [reply Client/Frontend/Fakespot/FakespotUtils.swift:88]: Ah missed them, ok thank you!

File: Client/Frontend/Toolbar+URLBar/TabLocationView.swift (hunk 1 above)
Comment (OrlaM) [reply Client/Frontend/Toolbar+URLBar/TabLocationView.swift]: Update LGTM, I'll let @yoanarios give it the final sign off.

**Review by yoanarios — APPROVED** [review]

**Review by lmarceau — APPROVED** [review]

**Comment by thatswinnie** [comment]
@Mergifyio backport release/v121

**Pull request merged**